alienfile

```diff
@@ -46,90 +46,100 @@ plugin 'PkgConfig' => (
+            '-DCMAKE_INSTALL_PREFIX:PATH=%{.install.prefix}',
+            '-DCMAKE_MAKE_PROGRAM:PATH=%{make}',
+        );
+
+        meta->before_hook(build => sub {
+            my($build) = @_;
+            my $prefix = $build->install_prop->{prefix};
+            $prefix =~ s{/}{\\}g;
+            meta->interpolator->add_helper(prefix_win => sub { $prefix });
+        });
+
+        meta->after_hook(gather_share => sub {
+            my $build= shift;
+            my $flags = '-DWIN32 -D_WIN32';
+            $flags .= ' -DWIN64 -D_WIN64' if $bits == 64;
+            if (my $ver = WINVER()) {
+                $flags .= " -D_WINVER=$ver -D_WIN32_WINNT=$ver"
+            }
+            $build->runtime_prop->{$_} .= " $flags" for qw( cflags cflags_static );
+            # on windows, we need the following libraries to be included. MinGW can't pull these
+            # from source on windows, so we add the equivalent to these pragma comments
+            # to our libs/libs_static area:
+            #pragma comment(lib, "Advapi32.lib")
+            #pragma comment(lib, "IPHLPAPI.lib")
+            #pragma comment(lib, "kernel32.lib")
+            #pragma comment(lib, "Psapi.lib")
+            #pragma comment(lib, "Shell32.lib")
+            #pragma comment(lib, "User32.lib")
+            #pragma comment(lib, "Userenv.lib")
+            #pragma comment(lib, "Ws2_32.lib")
+            $build->runtime_prop->{$_} .= ' -ladvapi32 -lIphlpapi -lkernel32 -lpsapi -lshell32 -luser32 -luserenv -lws2_32' for qw( libs libs_static );
+        });
+
+        build [
+            ['%{cmake}', @args, '%{.install.extract}' ],
+            ['%{make}' ],
+            'mkdir %{prefix_win}\\lib',
+            'mkdir %{prefix_win}\\lib\\pkgconfig',
+            'copy libuv_a.a %{prefix_win}\\lib\\libuv.a',
+            'copy libuv.dll %{prefix_win}\\lib',
+            'copy libuv.dll.a %{prefix_win}\\lib',
+            'mkdir %{prefix_win}\\include',
+            'mkdir %{prefix_win}\\include\\uv',
+            'copy include\\*.h %{prefix_win}\\include',
+            'copy include\\uv\\*.h %{prefix_win}\\include\\uv',
+            sub {
+                my($build) = @_;
+                my($pc) = Path::Tiny->new('libuv.pc.in')->slurp;
+
+                my $prefix = $build->runtime_prop->{prefix};
+                my $version = $build->runtime_prop->{version};
+
+                $pc =~ s{\@prefix\@}{$prefix}g;
+                $pc =~ s{\@libdir\@}{$prefix/lib}g;
+                $pc =~ s{\@includedir\@}{$prefix/include}g;
+                $pc =~ s{\@PACKAGE_NAME\@}{libuv}g;
+                $pc =~ s{\@PACKAGE_VERSION\@}{$version}g;
+                $pc =~ s{\@LIBS\@}{}g;
+
+                Path::Tiny->new($build->install_prop->{prefix})->child('lib/pkgconfig/libuv.pc')->spew($pc);
+            },
+        ];
+    }
+    else {
+        requires 'Alien::Autotools';
+        plugin 'Build::Autoconf' => ();
+
+        build [
+            'sh autogen.sh',
```

Comment:
I forgot that we needed autotools and not just sh and friends.  I am a little more ambivalent about autoconf vs cmake.